README.md

```diff
@@ -1,85 +1,86 @@
-impute or interpolate values for any missing fields in the data set. For more information on these strategies, pass the 
---help option to this script:
+Because observed weather data commonly contains gaps, DIYEPW will attempt to fill in any such gaps to ensure that in 
+each record a value is present for all of the hourly timestamps for the variables shown above. To do so, it will use one 
+of two strategies to impute or interpolate values for any missing fields in the data set:
 
-```
-python create_amy_epw.py --help
-```
+#### Interpolation: Handling for small gaps
+Small gaps (by default up to 6 consecutive hours of consecutive missing data for a field), are handled by linear 
+interpolation, so that for example if the dry bulb temperature has a gap with neighboring observed values like 
+(20, X, X, X, X, 25), DIYEPW will replace the missing values to give (20, 21, 22, 23, 24, 25).
 
-The maximum number of rows that may be imputed or interpolated can also be set by passing options to the script:
+#### Imputation: Handling for large gaps
+Large gaps (by default up to 48 consecutive hours of missing data for a field) are filled using an imputation strategy
+whereby each missing field is set to the average of the field's value two weeks in the past and in the future from
+the missing timestamp.
 
-```
-python create_amy_epw.py --max-records-to-interpolate=6 --max-records-to-impute=48
-```
+If a gap exists in the data that is larger than the maximum allowed for the imputation strategy, that file will be
+rejected and no EPW file will be generated.
 
-This script will also convert these quantities from NOAA units to units consistent with those expected in an EPW file, 
-as shown in [the EPW data dictionary](https://bigladdersoftware.com/epx/docs/8-3/auxiliary-programs/energyplus-weather-file-epw-data-dictionary.html).
+The maximum number of missing values that the interpolation strategy will be used for, and the maximum number of
+missing values that can be imputed, can be changed from their defaults. The functions that generate EPW files, 
+`create_amy_epw_file()` and `create_amy_epw_files_for_years_and_wmos()`, both accept the optional arguments
+`max_records_to_interpolate` and `max_records_to_impute`, which likewise override the defaults of 6 and 48.
   
-## Reading in TMY3 files and writing EPW files
-Functions for reading TMY3 files and writing EPW files within this script were taken or adapted from the 
-[LAF.py script](https://github.com/SSESLab/laf/blob/master/LAF.py) by Carlo Bianchi at the Site-Specific Energy Systems 
-Lab repository.
+## Package Functions
+All of the functionality of the DIYEPW project is available as a set of functions that underlie the scripts 
+described above. The functions offer much more granular access to the capabilities of the project, and allow
+DIYEPW capabilites to be incorporated into other software projects.
+
+The functions provided by the package are as follows:
+
```

Comment:
The formatting for the below isn't picked up quite right by github (might just need extra lines between functions) -- see https://github.com/IMMM-SFA/diyepw/tree/csv_based_script#package-functions

Also, would be nice to have the function signatures described as well

Reply:
I agree, this doesn't look good rendered. I've changed the functions to be a bullet list instead, which I think looks a lot nicer.